SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp

```diff
@@ -754,14 +754,31 @@ bool FSpatialGDKEditorToolbarModule::FetchInspectorBinaryWrapper(FString Inspect
 {
 	bFetchingInspectorBinary = true;
 
-	const bool bSuccess =
-		SpatialCommandUtils::FetchInspectorBinary(InspectorVersion, GetDefault<USpatialGDKSettings>()->IsRunningInChina());
+	bool bSuccess = SpatialCommandUtils::FetchInspectorBinary(InspectorVersion, GetDefault<USpatialGDKSettings>()->IsRunningInChina());
 
 	if (!bSuccess)
 	{
 		UE_LOG(LogSpatialGDKEditorToolbar, Error, TEXT("Could not fetch the Inspector for version %s"), *InspectorVersion);
 		OnShowFailedNotification(TEXT("Failed to fetch local inspector!"));
+
+		bFetchingInspectorBinary = false;
+		return false;
+	}
+
+#if PLATFORM_MAC
+		int32 OutCode = 0;
+	FString OutString;
+	FString OutErr;
+	FString ChmodCommand = FPaths::Combine(SpatialGDKServicesConstants::BinPath, TEXT("chmod"));
+	FString ChmodArguments = FString::Printf(TEXT("+x %s"), *SpatialGDKServicesConstants::GetInspectorExecutablePath(InspectorVersion));
+	bSuccess = FPlatformProcess::ExecProcess(*ChmodCommand, *ChmodArguments, &OutCode, &OutString, &OutErr);
```

Comment:
How come we only need to do this for the inspector and not the runtime binary? Or do we actually do this for the runtime binary elsewhere?

Reply:
I'm not entirely sure. it could be due to how we start up the runtime and the inspector or how the packages are stored in the first place. the runtime seems to start up without any problems

Reply:
Ah it looks like that may be because the runtime package is retrieved as an archive and then unzipped, so that probably automatically sets the right permissions, and for the inspector the package itself is the binary.